src/lib/data/history.ts

```diff
@@ -0,0 +1,87 @@
+import {db, Q, HistoryModel, RideModel} from '../../database';
+import {
+  TrainingCenterDatabase,
+  Activity,
+  ActivityLap,
+  Track,
+  TrackPoint,
+  Position,
+  HeartRateBpm,
+  TrackPointExtensions,
+  ActivityList,
+  HeartRateInBeatsPerMinute,
+} from 'tcx-builder';
+import {getRideAggregates} from './aggregates';
+
+export async function getRideHistory(ride: RideModel) {
+  return db.get<HistoryModel>('history').query(Q.where('ride_id', ride.id));
+}
+
+export async function generateTCX(ride: RideModel) {
+  if (!ride.endedAt) {
+    throw new Error(
+      `Ride ${ride.id} is not complete - can't generate tcx file.`,
+    );
+  }
+
+  const records = await getRideHistory(ride);
+  const aggregates = await getRideAggregates(ride);
+
+  const trackPoints: TrackPoint[] = [];
+  for (const record of records) {
+    trackPoints.push(
+      new TrackPoint({
+        time: new Date(record.createdAt),
+        ...(record.latitude &&
+          record.longitude && {
+            position: new Position(record.latitude, record.longitude),
+          }),
+        ...(record.altitude && {altitudeMeters: record.altitude}),
+        // not sure if they want distance between points or total
+        distanceMeters: record.distance,
+        ...(record.heartRate && {
+          heartRateBpm: new HeartRateBpm(record.heartRate),
+        }),
+        ...(record.cadence && {cadence: record.cadence}),
+        sensorState: 'Present',
+        ...(record.speed && {
+          extensions: new TrackPointExtensions({
+            Speed: record.speed,
+          }),
+        }),
+      }),
+    );
+  }
+
+  const myLap: ActivityLap = new ActivityLap(new Date(ride.startedAt), {
+    Calories: 0,
+    Intensity: 'Active',
+    TriggerMethod: 'Distance',
```

Comment:
Think we should change the trigger method to `manual` since that means it was started with button press. These are the options and what they mean:

- Manual: This means the lap was triggered manually by the user, typically by pressing a button on their GPS device or sports watch.
- Time: This means the lap was triggered automatically based on a predefined time interval (e.g., every 10 minutes).
- Distance: This means the lap was triggered automatically based on a predefined distance interval (e.g., every 1 kilometer or mile).
- Location: This means the lap was triggered automatically based on a specific location or waypoint.